### PR TITLE
[build] fix inconsistent Kafka shading among cdc connectors

### LINKED ISSUE #2984: [Bug] Inconsistent Kafka shading among cdc connectors
### Search before asking

- [X] I searched in the [issues](https://github.com/ververica/flink-cdc-connectors/issues) and found nothing similar.

### Flink version

1.14.3

### Flink CDC version

3.0.0

### Database and its version

MySQL 5.7
Mongo 4.4

### Minimal reproduce step

Put flink-sql-connector-tidb-cdc-3.0.0.jar, flink-sql-connector-mysql-cdc-3.0.0.jar and flink-sql-connector-mongodb-cdc-3.0.0.jar into lib folder at Flink home (Flink 1.14 in my case), then create a MySQL CDC table and submit a simple query like select * from mysql_cdc_table, there would be an error:

`java.lang.NoClassDefFoundError: org/apache/kafka/connect/source/SourceRecord`

It's caused by inconsistent shading of  Kafka classes. Only MySQL CDC connector relocated Kafka classes, so if the classpath contains un-relocated classes from other CDC connectors the above error occurs.

### What did you expect to see?

The cdc connectors would not have class conflicts.

### What did you see instead?

The cdc connectors have class conflicts.

### Anything else?

_No response_

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by link3280**
As titled.

Close #2984 .

### CODE REVIEW

**Comment by link3280** [comment]
cc @leonardBang . Should a quick one.

**Review by lvyanquan — APPROVED** [review]

**Comment by link3280** [comment]
ping @leonardBang

File: flink-cdc-connect/flink-cdc-source-connectors/flink-sql-connector-tidb-cdc/pom.xml [hunk 1]
```diff
@@ -60,6 +60,12 @@ under the License.
                                 </includes>
                             </artifactSet>
                             <relocations>
+                                <relocation>
+                                    <pattern>org.apache.kafka</pattern>
+                                    <shadedPattern>
+                                        com.ververica.cdc.connectors.shaded.org.apache.kafka
```
Comment (leonardBang) [inline flink-cdc-connect/flink-cdc-source-connectors/flink-sql-connector-tidb-cdc/pom.xml:66]: Maybe we should change `  com.ververica.`  to `org.apache.flink` after we donated to Apache?

Comment (link3280) [reply flink-cdc-connect/flink-cdc-source-connectors/flink-sql-connector-tidb-cdc/pom.xml:66]: You're right. I didn't notice the package name had changed. Hold on a sec.

Comment (link3280) [reply flink-cdc-connect/flink-cdc-source-connectors/flink-sql-connector-tidb-cdc/pom.xml:66]: I found no shaded pattern has migrated from `com.ververica.cdc.connectors.shaded` yet. Maybe we should do it all together in an other PR?

Comment (Shawn-Hx) [reply flink-cdc-connect/flink-cdc-source-connectors/flink-sql-connector-tidb-cdc/pom.xml:66]: > I found no shaded pattern has migrated from `com.ververica.cdc.connectors.shaded` yet. Maybe we should do it all together in an other PR?

See https://github.com/apache/flink-cdc/pull/3110.

Comment (link3280) [reply flink-cdc-connect/flink-cdc-source-connectors/flink-sql-connector-tidb-cdc/pom.xml]: The shaded pattern is updated. Thanks.

**Comment by link3280** [comment]
cc @PatrickRen

**Review by Shawn-Hx — APPROVED** [review]

**Review by PatrickRen — APPROVED** [review]
@link3280 Thanks for the patch! LGTM

**Pull request merged**